### DISCUSSION TITLE
Sonoff THR320D configuration challenges

### DISCUSSION THREAD

**Discussion by badhras** [arendst/Tasmota#17698] [Q&A] [answered]
I was able to flash tasmota32.bin to my Sonoff THR320D and configure the WiFi. I also have the THS01 sensor attached. Unfortunately, I'm not seeing the Toggle buttons nor the temperature readings on the Tasmota **Main** page. The LCD screen is also blank. I appreciate any advice for what I should try next. This is my first time flashing Tasmota firmware to a device, although I have configured MartinJerry devices that were pre-flashed with Tasmota.

The **Information** page shows the following.
Program Version	12.3.1.3(tasmota)
Build Date & Time	2023-01-13T16:23:35
Core/SDK Version	2_0_6/v4.4.3.20221227

I did a copy-modify-paste on the **Configure Other** page's **Template** field.
`{"NAME":"Sonoff THR320D","GPIO":[32,0,0,0,226,9280,0,0,0,321,0,576,320,9184,9216,9312,0,0,9313,9248,0,1,0,3840,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1}
`
I used https://templates.blakadder.com/sonoff_THR320D.html and copied the string verbatim.

On the **Configure Template** page, I see:
```
Name:       THR320D
Based on:   ESP32-Devkit (1)
GPIO0:      Button 1
GPIO1-3:    None
GPIO4:      Relay 3
GPIO5:      TM1621 DAT
GPIO6-11:   None, the GPIO text is colored red
GPIO12:     None
GPIO13:     Led_i 2
GPIO14:     None
GPIO15:     LedLink_i
GPIO16:     Led_i 1, the GPIO16 text is colored red
GPIO17:     TM1621 CS, the GPIO17 text is colored red
GPIO18:     TM1621 WR
GPIO19:     Relay_b 1
GPIO20-21:  None
GPIO22:     Relay_b 2
GPIO23:     TM1621 RD
GPIO24:     None
GPIO25:     SI7021
GPIO26:     None
GPIO27:     Output Hi
GPIO32-39:  None
```

I used SI7021 for GPIO25, based on this page. https://techestigate.com/sonoff-th-elite-to-tasmota/

I have also tried this template string:
`{"NAME":"Sonoff THR320D","GPIO":[32,0,0,0,226,9280,0,0,0,321,0,576,320,9184,9216,9312,0,0,9313,9248,0,1,0,3840,0,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":1} 
`
which I obtained from here: https://github.com/arendst/Tasmota/blob/development/TEMPLATES.md

This did not change any behavior from the main page. I don't see any buttons or temperature readings. It did cause this change in the **Configure Templates** page.
`GPIO25:     User `

**Comment by barbudor** [comment, marked as answer]
Did you activate the template?
(check the box below the template in configure others)

**Reply by badhras** [reply]
Thanks for the suggestion. That certainly fixed it. I somehow completely missed that step, and I never noticed the box.

**Reply by sfromis** [reply]
When it comes to getting it to work with one of the available sensor probes, the intention is that you can use auto-configuration, and in the dropdown find a setup tailored to the device and probe you have, instead of working out how to get gpio25 set correctly.

**Comment by cju-mt** [comment]
How do you change it to fahrenheit. I don't see a way to change it. Maybe I am just missing it...

**Comment by cju-mt** [comment]
Something strange that happend. I had to apply the auto configuration twice to get the sensors to come up. I am using the THS01/SI7021

**Comment by cju-mt** [comment]
Just figured it out and in case this will help anybody else. The easiest way to set it is from the web UI console

See https://tasmota.github.io/docs/Commands/#setoptions

type

SetOption8 1

press enter and it should say somethin similar to. 

1:05:55.279 RSL: RESULT = {"SetOption8":"ON"}

**Comment by jmcucurella** [comment]
When using THR320D as a thermostat, all is working ok, but deadband is 5 celsius degree. Some one knows to reduce this deadband ?

**Reply by sfromis** [reply]
Are you using the Tasmota [Thermostat](https://tasmota.github.io/docs/Thermostat/) feature? In that case, there are a lot of [commands](https://tasmota.github.io/docs/Commands/#thermostat) for managing temperature handling.

**Reply by jmcucurella** [reply]
By default, THR320D with tasmota works as a thermostat.
I don't know how to implement a specific thermostat as you say. Probably with this function, and modifying PropBandSet and TimeResetSet, I can get better performance.
Could you explain how implement specific Thermostat function. 
I'm using it with Home Assistant.
Any help will be appreciated.

**Reply by sfromis** [reply]
From what I see, there is no default to setup any thermostat feature, when installing Tasmota on the THR320D, such depends on what you specifically did to config the device.

Anyway, not having worked with the Tasmota Thermostat feature (assuming that this is actually what you're using), and not using HA either (maybe you have some thermostat feature activated there), I cannot explain anything, other than refer you to explanations in the docs.

**Reply by jmcucurella** [reply]
Ok, I will check if  thermostat is activated on Tasmota
